### PR TITLE
Enabling Context Menu on Keyboard input in UWP

### PR DESCRIPTION
**Pull request by roopasp**
Enabling Context Menu to show onKeydown when the Application button or shift+F10 are pressed. Enabled for View, Button and Link types. The context menu is displayed offset to the top left corner of the element on which it is called.

### CODE REVIEW

File: src/native-common/utils/EventHelpers.ts [hunk 1]
```diff
@@ -59,6 +59,12 @@ export class EventHelpers {
                         keyCode = 18;
                         break;
 
+                    //keyCode in windows is 93 for context menu button. Since that is already used
```
Comment (erictraut) [inline src/native-common/utils/EventHelpers.ts]: Nit: Please add spaces after "//" for consistency.

File: src/windows/Button.tsx [hunk 2]
```diff
@@ -45,6 +47,11 @@ export class Button extends ButtonBase implements React.ChildContextProvider<But
 
     private _isFocusedWithKeyboard = false;
 
+    //Offset to show context menu using keyboard.
```
Comment (erictraut) [inline src/windows/Button.tsx]: Nit: please add space after "//" for consistency.

File: src/windows/Button.tsx [hunk 3]
```diff
@@ -45,6 +47,11 @@ export class Button extends ButtonBase implements React.ChildContextProvider<But
...
+    protected _contextMenuOffset() {
```
Comment (erictraut) [inline src/windows/Button.tsx]: Method names should all begin with a verb. This should be _getContextMenuOffset().

File: src/windows/Link.tsx [hunk 4]
```diff
@@ -32,6 +35,12 @@ export interface LinkState {
 }
 
 export class Link extends LinkBase<LinkState> implements FocusManagerFocusableComponent {
+    
+    //Offset to show context menu using keyboard.
```
Comment (erictraut) [inline src/windows/Link.tsx]: Nit: Please add space after "//" for consistency.

File: src/windows/Link.tsx [hunk 5]
```diff
@@ -32,6 +35,12 @@ export interface LinkState {
...
+    protected _contextMenuOffset() {
```
Comment (erictraut) [inline src/windows/Link.tsx]: See comment above. This should be named _getContextMenuOffset().

File: src/windows/Link.tsx [hunk 6]
```diff
@@ -32,6 +35,12 @@ export interface LinkState {
...
+        return {x: 0, y: 0};
```
Comment (erictraut) [inline src/windows/Link.tsx]: Nit: there should be extra spaces within brackets.

File: src/windows/Button.tsx [hunk 7]
```diff
@@ -45,6 +47,11 @@ export class Button extends ButtonBase implements React.ChildContextProvider<But
...
+        return {x: 0, y: 0};
```
Comment (erictraut) [inline src/windows/Button.tsx]: Nit: there should be extra spaces within the curly brackets for consistency.

File: src/windows/View.tsx [hunk 8]
```diff
@@ -60,6 +63,11 @@ export class View extends ViewCommon implements React.ChildContextProvider<ViewC
         ...ViewCommon.childContextTypes
     };
 
+    //Offset to show context menu using keyboard.
```
Comment (erictraut) [inline src/windows/View.tsx]: See above. Add spaces after "//", fix name of method, and add spaces within curly brackets.

File: src/windows/View.tsx [hunk 9]
```diff
@@ -182,17 +205,30 @@ export class View extends ViewCommon implements React.ChildContextProvider<ViewC
...
-            // called from super constructors are involved.
-            if (!this._onKeyDown) {
-                this._onKeyDown =  (e: Types.SyntheticEvent) => {
-                    if (this.props.onKeyPress) {
-                        // A conversion to a KeyboardEvent looking event is needed
-                        this.props.onKeyPress(EventHelpers.toKeyboardEvent(e));
-                    }
-                };
+            // called from super constructors are involved. Ensuring nothing happens here if there 
+            // is a tabIndex else KeyDown is handled twice, in _onFocusableKeyDown as well.
```
Comment (erictraut) [inline src/windows/View.tsx]: Do you mean "if a tabIndex is specified" rather than "is a tabIndex"?

Comment (roopasp) [reply src/windows/View.tsx]: yes, fixed.

File: src/windows/View.tsx (hunk 8 above)
Comment (roopasp) [reply src/windows/View.tsx]: updated as per conventions

File: src/windows/Button.tsx (hunk 7 above)
Comment (roopasp) [reply src/windows/Button.tsx]: updated

File: src/windows/Link.tsx (hunk 6 above)
Comment (roopasp) [reply src/windows/Link.tsx]: updated

File: src/windows/Link.tsx (hunk 5 above)
Comment (roopasp) [reply src/windows/Link.tsx]: updated

File: src/windows/Link.tsx (hunk 4 above)
Comment (roopasp) [reply src/windows/Link.tsx]: updated

File: src/windows/Button.tsx (hunk 3 above)
Comment (roopasp) [reply src/windows/Button.tsx]: Fixed naming.

File: src/windows/Button.tsx (hunk 2 above)
Comment (roopasp) [reply src/windows/Button.tsx]: Added in all comments.

File: src/native-common/utils/EventHelpers.ts (hunk 1 above)
Comment (roopasp) [reply src/native-common/utils/EventHelpers.ts]: Added in all comments

**Comment by erictraut** [comment]
Looks good to me. I've asked @reseul to review it also.

**Pull request merged**